apprise/plugins/NotifyHomeAssistant.py

```diff
@@ -0,0 +1,310 @@
+        'user': {
+            'name': _('Username'),
+            'type': 'string',
+        },
+        'password': {
+            'name': _('Password'),
+            'type': 'string',
+            'private': True,
+        },
+        'accesstoken': {
+            'name': _('Long-Lived Access Token'),
+            'type': 'string',
+            'private': True,
+            'required': True,
+        },
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'nid': {
+            # Optional Unique Notification ID
+            'name': _('Notification ID'),
+            'type': 'string',
+            'regex': (r'^[a-f0-9_-]+$', 'i'),
+        },
+    })
+
+    def __init__(self, accesstoken, nid=None, **kwargs):
+        """
+        Initialize Home Assistant Object
+        """
+        super(NotifyHomeAssistant, self).__init__(**kwargs)
+
+        self.fullpath = kwargs.get('fullpath', '')
+
+        if not (self.secure or self.port):
+            # Use default insecure port
+            self.port = self.default_insecure_port
+
+        # Long-Lived Access token (generated from User Profile)
+        self.accesstoken = validate_regex(accesstoken)
+        if not self.accesstoken:
+            msg = 'An invalid Home Assistant Long-Lived Access Token ' \
+                  '({}) was specified.'.format(accesstoken)
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # An Optional Notification Identifier
+        self.nid = None
+        if nid:
+            self.nid = validate_regex(
+                nid, *self.template_args['nid']['regex'])
+            if not self.nid:
+                msg = 'An invalid Home Assistant Notification Identifier ' \
+                      '({}) was specified.'.format(nid)
+                self.logger.warning(msg)
+                raise TypeError(msg)
+
+        return
+
+    def send(self, body, title='', notify_type=NotifyType.INFO, **kwargs):
+        """
+        Sends Message
+        """
+
+        # Prepare our persistent_notification.create payload
+        payload = {
+            'title': title,
+            'message': body,
+            # Use a unique ID so we don't over-write the last message
+            # we posted. Otherwise use the notification id specified
+            'notification_id': self.nid if self.nid else str(uuid4()),
+        }
+
+        # Prepare our headers
+        headers = {
+            'User-Agent': self.app_id,
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer {}'.format(self.accesstoken),
+        }
+
+        auth = None
+        if self.user:
+            auth = (self.user, self.password)
+
+        # Set our schema
+        schema = 'https' if self.secure else 'http'
+
+        url = '{}://{}'.format(schema, self.host)
+        if isinstance(self.port, int):
+            url += ':%d' % self.port
+
+        url += '/' + self.fullpath.strip('/')
```

Comment:
@caronc you have the `url` method below. it looks like you are constructing the `url` manually here. why don't you use the `url` method instead?

I seem to have run into an issue here. when the `self.fullpath` is empty, the url becomes like `//api/services/persistent_notification/create`, which causes the error
```
Socket Exception: HTTPConnectionPool(host='localhost', port=8123): Max retries exceeded with url: //api/services/persistent_notification/create (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0xffff8b9d0730>, 'Connection to localhost timed out. (connect timeout=4.0)'))
```
My example URLs are:
```
hassio://localhost/{access_token} \
hassio://localhost:8123/{access_token} \
```

Reply:
Good find; I'll get this fixed.